enhancements/console/internationalization.md

```diff
@@ -0,0 +1,135 @@
+language (i.e. Chinese or Japanese) rather than English.
+
+Using the i18next library, we will have a single source code for all languages of the product and it will also be easier to
+leverage content strategists to author content and review copy consistency. All translatable strings in the front-end will be marked in the code and will be easy to export and modify.
+
+### Goals
+
+* Mark and extract to-be-translated strings in console
+* Work with the globalization team to obtain translations and perform localization testing for Chinese and Japanese
+* Work with UX to drive copy consistency in the UI (confirmation messages, actions that are more than one word, pieces from CoreOS, and similar actions with different verbiage)
+
+### Non-Goals
+
+We are not able to translate all text in the application. Text located in non-Red-Hat-controlled development environments may not be accessible for translation.
+This may include items such as events surfaced from Kuberenetes, alerts, and error messages displayed to the user or in logs.
+In addition, operators that surface informational messages may not have translations available, and localization of logging messages at any level is not in scope.
+Localization will not be included in the CLI at this time.
+
+## Proposal
+
+[i18next](https://www.i18next.com/) is a Javascript internationalization framework.
+It provides standard i18n features such as plurals, context, interpolation, and format,
+as well as language detection, loading, caching, and extensions.
+
+We have already [created a PR](https://github.com/openshift/console/pull/6194) integrating the i18next library into OpenShift.
+We have also been working with the Red Hat Globalization team to test out the hand-off process.
+
+We will need to mark all hard-coded strings with i18next so they can be extracted using an i18next extension and handed off to the Globalization team.
+This is not complex work but will involve many small code changes.
+[A rough string search](https://gist.github.com/jschuler/a404aed1ea774383cc2decd57d3135c6)
+of the frontend console code found about 9,000 strings, but many will not be translated (such as Kubernetes resource names).
+We will need to identify a prioritized list of UIs in OpenShift for localization and
+determine fallback behavior for pieces of the UI that are not translated.
+
+We will also need to do testing, coordinate with the Red Hat Globalization team on testing and translation,
+and collaborate with console plugins on a localization strategy for releases. Finally, a product testing window will need to be defined.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Localization will affect multiple teams within the console development space.
+This will affect Dev console, Admin console, and other packages like CNV.
+The solution must account for the plugin contribution model and enable dynamic extensions to provide localization as necessary.
```

Comment:
We'll need to explore how this will work with Webpack Module Federation for dynamic plugins, but I think that can be a follow up PR. We have a story tracking it.

Reply:
Can you mention that we plan to break translations up into multiple translation files, at least one per package?

https://react.i18next.com/guides/multiple-translation-files